parquet-hadoop/src/main/java/org/apache/parquet/HadoopReadOptions.java

```diff
@@ -52,18 +55,23 @@ private HadoopReadOptions(boolean useSignedStringMinMax,
                             boolean useColumnIndexFilter,
                             boolean usePageChecksumVerification,
                             boolean useBloomFilter,
+                            boolean enableAsyncIOReader,
```

Comment:
life would be a lot simpler if this code moved to a parameter object containing all the args: adding new options would be straightforward. We did this for the S3ClientFactory to stop external implementations having incompatible signatures when new stuff went in